test/contracts/oft/v2/NativeOFTV2.test.js

```diff
@@ -0,0 +1,454 @@
+
+        expect(await localEndpoint.getChainId()).to.equal(localChainId)
+        expect(await remoteEndpoint.getChainId()).to.equal(remoteChainId)
+
+        //------  deploy: base & other chain  -------------------------------------------------------
+        // create two NativeOFTV2 instances. both tokens have the same name and symbol on each chain
+        // 1. base chain
+        // 2. other chain
+        nativeOFTV2 = await NativeOFTV2.deploy(name, symbol, sharedDecimals, localEndpoint.address)
+        remoteOFTV2 = await OFTV2.deploy(name, symbol, sharedDecimals, remoteEndpoint.address)
+
+        // internal bookkeeping for endpoints (not part of a real deploy, just for this test)
+        await localEndpoint.setDestLzEndpoint(remoteOFTV2.address, remoteEndpoint.address)
+        await remoteEndpoint.setDestLzEndpoint(nativeOFTV2.address, localEndpoint.address)
+
+        //------  setTrustedRemote(s) -------------------------------------------------------
+        // for each OFTV2, setTrustedRemote to allow it to receive from the remote OFTV2 contract.
+        // Note: This is sometimes referred to as the "wire-up" process.
+        // set each contracts source address so it can send to each other
+        remotePath = ethers.utils.solidityPack(["address", "address"], [remoteOFTV2.address, nativeOFTV2.address])
+        localPath = ethers.utils.solidityPack(["address", "address"], [nativeOFTV2.address, remoteOFTV2.address])
+        await nativeOFTV2.setTrustedRemote(remoteChainId, remotePath)
+        await remoteOFTV2.setTrustedRemote(localChainId, localPath)
+
+        await nativeOFTV2.setUseCustomAdapterParams(true)
+        // ... the deployed OFTs are ready now!
+    })
+
+    it("sendFrom() - tokens from main to other chain using default", async function () {
+        expect(await ethers.provider.getBalance(localEndpoint.address)).to.be.equal(ethers.utils.parseUnits("0", 18))
+
+        await nativeOFTV2.setUseCustomAdapterParams(false)
+        await remoteOFTV2.setUseCustomAdapterParams(false)
+
+        // ensure they're both allocated initial amounts
+        let aliceBalance = await ethers.provider.getBalance(alice.address)
+        expect(await nativeOFTV2.balanceOf(owner.address)).to.equal(0)
+        expect(await remoteOFTV2.balanceOf(owner.address)).to.equal(0)
+        expect(await ethers.provider.getBalance(nativeOFTV2.address)).to.be.equal(0)
+
+        let depositAmount = ethers.utils.parseUnits("7", 18)
+        await nativeOFTV2.deposit({ value: depositAmount })
+
+        expect(await nativeOFTV2.balanceOf(owner.address)).to.equal(depositAmount)
+        expect(await ethers.provider.getBalance(nativeOFTV2.address)).to.be.equal(depositAmount)
+
+        let leftOverAmount = ethers.utils.parseUnits("0", 18)
+        let totalAmount = ethers.utils.parseUnits("8", 18)
+
+        const ownerAddressBytes32 = ethers.utils.defaultAbiCoder.encode(["address"], [owner.address])
```

Comment:
`Bytes32` encoding can be done in `beforeEach` section rather than in each individual test